### PR TITLE
docs: add `at` for array and typeArray and add some other methods for typedArray

### PR DESCRIPTION
**Pull request by yjhmelody**
The descriptions are from index.d.ts.

### CODE REVIEW

**Comment by yjhmelody** [comment]
Hi @MaxGraey  Done what your just said.

File: src/stdlib/typedarray.md [hunk 1]
```diff
@@ -137,7 +142,26 @@ The TypedArray API works very much like JavaScript's \([MDN](https://developer.m
   function reverse(): this
   ```
   Reverses an array's values in place, modifying the array before returning it.
+* ```ts
+  join(separator?: string): string;
```
Comment (MaxGraey) [inline src/stdlib/typedarray.md]:
```suggestion
  function join(separator?: string): string;
```

File: src/stdlib/typedarray.md [hunk 2]
```diff
@@ -137,7 +142,26 @@ The TypedArray API works very much like JavaScript's \([MDN](https://developer.m
...
+  ```
+  The join() method joins all elements of an array into a string. This method has the same algorithm as Array.prototype.join().
 
+* ```ts
+  function slice(start?: i32, end?: i32): TypedArray
+  ```
+  Returns a shallow copy of the array's values from begin inclusive to end exclusive, as a new array. If omitted, end defaults to the end of the array.
+
+* ```ts
+  copyWithin(target: i32, start: i32, end?: i32): this;
```
Comment (MaxGraey) [inline src/stdlib/typedarray.md]:
```suggestion
  function copyWithin(target: i32, start: i32, end?: i32): this;
```

File: src/stdlib/typedarray.md [hunk 3]
```diff
@@ -137,7 +142,26 @@ The TypedArray API works very much like JavaScript's \([MDN](https://developer.m
...
+  ```
+  The copyWithin() method copies the sequence of array elements within the array to the position starting at target. The copy is taken from the index positions of the second and third arguments start and end. The end argument is optional and defaults to the length of the array.
+
+* ```ts
+  filter(callbackfn: (value: T, index: i32, self: this) => bool): this;
```
Comment (MaxGraey) [inline src/stdlib/typedarray.md]:
```suggestion
  function filter(
    fn: (value: T, index: i32, self: TypedArray) => bool
  ): TypedArray;
```

**Comment by yjhmelody** [comment]
Done :)

File: src/stdlib/typedarray.md [hunk 4]
```diff
@@ -137,7 +142,28 @@ The TypedArray API works very much like JavaScript's \([MDN](https://developer.m
   function reverse(): this
   ```
   Reverses an array's values in place, modifying the array before returning it.
+* ```ts
+  function join(separator?: string): string;
+  ```
+  The join() method joins all elements of an array into a string. This method has the same algorithm as Array.prototype.join().
```
Comment (MaxGraey) [inline src/stdlib/typedarray.md]: instead `The <some-method>() method ...` pattern better follow previous convention. See `reduceRight` and etc when description start from "Calls ..." or `Joins all elements of an array into a string. ` for current case. Plz do the same for rest new methods

Comment (yjhmelody) [reply src/stdlib/typedarray.md]: Yes, I know that. However, two documents (index.d.ts and website) need to be maintained for the same method, and the description styles are different, which is actually quite troublesome.

Anyway, I will modify it first

**Comment by yjhmelody** [comment]
Done :)

**Pull request closed without merging**